gravitee-apim-gateway/gravitee-apim-gateway-tests-sdk/README.adoc

```diff
@@ -0,0 +1,247 @@
+
+Default configuration uses a dynamic port for HTTP and another one for HTTPS.
+
+Here is an example allowing to configure secured endpoints:
+```java
+@Override
+protected void configureWireMock(WireMockConfiguration configuration) {
+    configuration
+        .needClientAuth(true)
+        .trustStorePath(ResourceUtils.toPath("certs/truststore01.jks"))
+        .trustStorePassword("password")
+        .keystorePath(ResourceUtils.toPath("certs/keystore01.jks"))
+        .keystorePassword("password");
+}
+```
+
+Then, you can simply use the instance of `wiremock` in your tests:
+```java
+wiremock.stubFor(get("/endpoint").willReturn(ok("A plain text body")));
+```
+
+TIP: For more information about wiremock, you can follow this https://wiremock.org/docs/stubbing/[documentation].
+
+==== Configure gateway
+
+Override `configureGateway(GatewayConfigurationBuilder gatewayConfigurationBuilder)` to be able to configure your gateway by passing properties to it.
+
+Here is an example to configure Gateway for HTTP2:
+```java
+gatewayConfigurationBuilder.set("http.secured", "true").set("http.alpn", "true").set("http.ssl.keystore.type", "self-signed");
+```
+
+==== Configure WebClient options
+
+To configure the `WebClient` injected as a parameter in a test method, you have to declare a `options` field using `@WebClientOptionsInject`.
+
+Here is an example to configure WebClient to do an HTTP2 call:
+
+```java
+@WebClientOptionsInject
+public WebClientOptions options = new WebClientOptions()
+    .setDefaultHost("localhost")
+    .setDefaultPort(gatewayPort())
+    .setSsl(true)
+    .setVerifyHost(false)
+    .setTrustAll(true);
+```
+
+TIP: As you can see, you can use `gatewayPort()` to get the configured gateway port.
+
+==== Configure Api
+
+Override `configureApi(Api api)` (implementation of ApiConfigurer) to configure the apis before their deployment. It's useful to add/update the configuration of all the apis being deployed in a programmatic way, instead of writing the same thing in the JSON definition.
+
+Let's say you want to modify an URL configured in a policy to use the wiremock port, you can do this:
+
+```java
+@Override
+public void configureApi(Api api) {
+    if (api.getId().equals("my-api-redirect")) {
+        api
+            .getFlows()
+            .forEach(flow -> {
+                flow
+                    .getPre()
+                    .stream()
+                    .filter(step -> policyName().equals(step.getPolicy()))
+                    .forEach(step ->
+                        step.setConfiguration(step.getConfiguration().replace(REDIRECT_URL, LOCALHOST + redirectServer.getPort()))
+                    );
+            });
+    }
+}
+```
+
+
+You can also use the field `deployedApis` to access and modify the deployed apis *at method level*.
+
+NOTE: It can be useful to change the state of the endpoints of the api, but you will not be able to modify a policy configuration on the fly.
```

Comment:
Not clear if it is never possible to modify a policy configuration on the fly or only if you've changed the state of the endpoints